### PR TITLE
[rfc][expo-updates][android] Remove legacy manifest support

### PR DESCRIPTION
**Pull request by wschurman**
# Why

This PR removes support from expo-updates for legacy manfiests.

I'm actually not sure we can do this because Expo Go currently uses expo-updates to load its manifests, and Expo Go will support legacy manifests for at least one more release? Curious what reviewers think.

Unless we make Expo Go single-versioned.

# How

Remove.

# Test Plan

Build.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-manifests/android/src/main/java/expo/modules/manifests/core/BareManifest.kt [hunk 1]
```diff
@@ -13,4 +15,29 @@ class BareManifest(json: JSONObject) : BaseLegacyManifest(json) {
 
   @Throws(JSONException::class)
   fun getCommitTimeLong(): Long = json.require("commitTime")
+
+  override fun getStableLegacyID(): String = json.getNullable("originalFullName") ?: getLegacyID()
+
+  override fun getScopeKey(): String = json.getNullable("scopeKey") ?: getStableLegacyID()
+
+  override fun getEASProjectID(): String? = json.getNullable("projectId")
```
Comment (ide) [inline packages/expo-manifests/android/src/main/java/expo/modules/manifests/core/BareManifest.kt:23]: I understand the intent of this PR is not to change the behavior of the BareManifest class and agree the current change looks good.

However, I'm wondering if fields like projectId, originalFullname, and sdkVersion are still top-level fields. (Maybe they are top-level fields in embedded manifests that are generated during builds?) It seems like a future PR may want to revisit these.

**Review by ide — COMMENTED** [review]
**Expo Go:** it depends on whether the removed code is versioned or not. If it is versioned, I see no problem removing it from Expo Go since older SDK versions will have versioned copies of this code.

Unfortunately, my reading of the code is that ExpoUpdatesAppLoader and the entire expo-manifests module are not versioned. I think your suspicion that this code might need to stay until Expo Go drops SDK 49 is correct.

Otherwise, the code looks good to me. However, I think it would be good to improve the errors that are thrown when accessing fields on the raw parsed JSON object since classic manifests will always be parsed as modern manifests. Error handling like: `if (looks like classic manifest) { throw error explaining the problem }` is easier for a developer to act on compared to a stack trace originating from `Manifest.getRawValue()`.

**Comment by wschurman** [comment]
Yep, I think it is unversioned. Going to close this, not much we can do for a year or so unless we change expo go to be only a single version (which we plan to do for release process anyways).

**Comment by ide** [comment]
As a final thought, what do you think about modifying `expo-updates` to throw a human-friendly error if it attempts to load a classic manifest with sdkVersion >= 50?

File: packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/ManifestFactory.kt [hunk 2]
```diff
@@ -11,14 +10,9 @@ import org.json.JSONObject
...
 object ManifestFactory {
-  private val TAG = ManifestFactory::class.java.simpleName
-
   @Throws(Exception::class)
   fun getManifest(manifestJson: JSONObject, responseHeaderData: ResponseHeaderData, extensions: JSONObject?, configuration: UpdatesConfiguration?): UpdateManifest {
     return when (val expoProtocolVersion = responseHeaderData.protocolVersion) {
-      null -> {
-        LegacyUpdateManifest.fromLegacyManifest(LegacyManifest(manifestJson), configuration!!)
-      }
       0, 1 -> {
```
Comment (wschurman) [inline packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/ManifestFactory.kt:16]: This means that in the past, 0 or 1 was always specified. Going forward, we can throw a nice error when null, especially if the manifest "looks like" a legacy manifest

**Pull request closed without merging**